Minor gcc warning in pdfgen.c

pdfgen.c: In function ‘find_word_break’:
pdfgen.c:1822:42: warning: comparison is always false due to limited range of data type [-Wtype-limits]
 1822 |     while (string && *string && (*string < 0 || !isspace(*string)))

---

static const char *find_word_break(const char *string)
{
    /* Skip over the actual word */
    while (string && *string && (*string < 0 || !isspace(*string)))
        string++;

    return string;
}

---

While statement should be:

    while (string && *string && (*string <= 0 || !isspace(*string)))

--- a/pdfgen.c
+++ b/pdfgen.c
@@ -1836,8 +1836,10 @@ int pdf_get_font_text_width(struct pdf_doc *pdf, const char *font_name,
 
 static const char *find_word_break(const char *string)
 {
+    if (!string)
+        return NULL;
     /* Skip over the actual word */
-    while (string && *string && (*string < 0 || !isspace(*string)))
+    while (*string && !isspace(*string))
         string++;
 
     return string;

